### DISCUSSION TITLE
Interested in Collaboration?

### DISCUSSION THREAD

**Discussion by jogibear9988** [GrapesJS/grapesjs#5813] [Ideas]
### Reproducible demo link

https://node-projects.github.io/web-component-designer-demo/index.html

### Describe the bug

I started a Framework to design webpages with webcomponents a few years ago, and now I found GrapeJS. Thought maybe we have the same goals and could benefit from each other? (Don't know how atm., but maybe) (link: https://node-projects.github.io/web-component-designer-demo/index.html)
The designer support of the flow layout is missing in mine at the moment, but it works very well with grid and flexbox, and also absolutely positioned elements. Also transformed (rotated) content works. And I've created a VSCode extension for it: https://marketplace.visualstudio.com/items?itemName=node-projects.vscode-designer-addon

So maybe there are parts where we could help out each other?

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by artf** [comment]
Interesting @jogibear9988 I'll take a look once I find some time 🥲

**Comment by jogibear9988** [comment]
I've now looked into your source, and used some parts of sorter.ts to start implementation of a placement for non absolute, non flexbox, non grid, wich mine was/is still missing.

I've now also looked into your studio, looks nice.
What my designer sample has, is support for usage of external webcomponents, maybe this would be interesting also for yours.

**Reply by artf** [reply]
Glad to hear that @jogibear9988 
GrapesJS should already be able to render external web components 🙂 you just need to configure the traits it order to be able to edit its properties.

**Reply by jogibear9988** [reply]
I've a parser for Custom Element Manifest files to display properties/events of custom elements (see: https://github.com/webcomponents/custom-elements-manifest).
Also they are installable directly from NPM into the running Designer.
Maybe this are also an Ideas for GrapeJS
